internal/indexer/implementation.go

```diff
@@ -348,7 +376,37 @@ func (i *Indexer) extractInterfacesAndConcreteTypes(pkgs []*packages.Package) (i
 		}
 	}
 
-	return interfaces, concreteTypes
+	batch := func(pkgBatch []string) error {
+		pkgs, err := i.loadPackage(true, pkgBatch...)
+		if err != nil {
+			return err
+		}
+
+		for _, pkg := range pkgs {
+			visit(pkg)
+		}
+		return nil
+	}
+
+	pkgBatch := []string{}
+	for ix, pkgName := range pkgNames {
+		pkgBatch = append(pkgBatch, pkgName)
+
+		if i.depBatchSize != 0 && ix%i.depBatchSize == 0 {
+			err := batch(pkgBatch)
+			runtime.GC() // Prevent a garbage pile
+			if err != nil {
+				return nil, nil, err
+			}
+			pkgBatch = []string{}
```

Comment:
```suggestion
			pkgBatch = pkgBatch[:0]
```

Reply:
Yeah it just resets the slice to the beginning of the array (all capacity, no length). Reuses the same memory over and over while it's live.

Reply:
Note that this trick only works if you're not using that array concurrently with something else that expects stable objects - this is fine since we're batching synchronously in a loop and not tasking things out to goroutines.